src/adapter/src/catalog/storage.rs

```diff
@@ -385,6 +412,151 @@ async fn migrate(
             )?;
             Ok(())
         },
+        // Object owners were added to object definitions.
+        //
+        // Introduced in v0.47.0
+        //
+        // TODO(jkosh44) Can be cleared (patched to be empty) in v0.50.0
+        |txn: &mut Transaction<'_>, now, _bootstrap_args| {
+            // Delete all system roles so we can re-insert them with known IDs.
+            let sys_roles = txn.roles.delete(|role_key, _| role_key.id.is_system());
+            assert_eq!(2, sys_roles.len(), "unexpected number of system roles");
+            txn.roles.insert(
+                RoleKey {
+                    id: MZ_SYSTEM_ROLE_ID,
+                },
+                RoleValue {
+                    role: (&*MZ_SYSTEM_ROLE).into(),
+                },
+            )?;
+            txn.roles.insert(
+                RoleKey {
+                    id: MZ_INTROSPECTION_ROLE_ID,
+                },
+                RoleValue {
+                    role: (&*MZ_INTROSPECTION_ROLE).into(),
+                },
+            )?;
+
+            // We need a default owner for all existing objects. So we just create a new role for
+            // this purpose named "default_owner". If one happens to exist with the same name, then
+            // just use that.
+            let default_owner_name = "default_owner";
+            let default_owner_id = match txn
+                .roles
+                .items()
+                .iter()
+                .filter(|(_, role_value)| role_value.role.name == default_owner_name)
+                .next()
+            {
+                Some((role_key, _)) => role_key.id,
```

Comment:
In the rare chance that some user somewhere has typed `CREATE ROLE default_owner`. I figured it's ok to use their existing role. If not, we can keep appending numbers to the end of `default_owner` until we find a unique role.

Reply:
I'm inclined to make the migration fail if this is the case so we can handle it by hand. There's essentially zero chance this will be a problem, and I'd feel more comfortable having knowledge that this is always a new role we are creating for this purpose.

Reply:
Why is failing the migration better than generating some unique role like `default_owner123`?

Reply:
Because then we can document what this role's name is without exception, and also don't have to have some code that probably won't run anyway, and don't have to write a test for it.

Reply:
This is also another one of those examples where it'd be super nice to have a PR step that is "if this PR merged, would it break any production deployment?" using stash-debug. There's difficult security and privacy concerns there, but it'd be nice.

Reply:
I've added a panic in this case.